k8s/zeppelin-server.yaml

```diff
@@ -115,7 +115,7 @@ spec:
             path: nginx.conf
       containers:
       - name: zeppelin-server
-        image: apache/zeppelin:0.9.0-SNAPSHOT
+        image: apache/zeppelin-server:0.9.0-SNAPSHOT
```

Comment:
Currently, the [official docker images](https://hub.docker.com/r/apache/zeppelin/tags) are built from `scripts/docker/zeppelin/bin/Dockerfile`.
Apache Infra configured the Dockerfile location for automated build on release. (see [this comment](https://issues.apache.org/jira/browse/INFRA-12781?focusedCommentId=16001034&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16001034))

Do you think we can release images based on new Dockerfiles in this PullRequest and remove `scripts/docker/zeppelin/bin/Dockerfile` ?

While `/k8s/zeppelin-server.yaml` points new docker image names, i think it make sense to release new images as well.

Reply:
I think we can delete `scripts/docker/zeppelin/bin/Dockerfile` for the next release. First we should check if Docker and k8s are able to use a flexible interpreter image. At least k8s is currently not able to use a flexible interpreter image.

It makes sense to push these new images. How should we handle different compilation versions?
At the moment I'm compiling Zeppelin with the [newest versions](https://github.com/apache/zeppelin/pull/3769/files#diff-3254677a7917c6c01f55212f86c57fbfR23) of Hadoop and Spark.
```
mvn -B package -DskipTests -Pbuild-distr -Pspark-3.0 -Phadoop3 -Pspark-scala-2.12 -Pweb-angular
```

Reply:
At least for Spark interpreter, it's got binary level compatibility to different spark (and hadoop) versions. Once built, It works with different versions without rebuilding it.

Reply:
> one image is more handy to work with

You are right, one or at least a small set of images is more practical for work. In fact, I currently have only three images (distribution image, server, (one large) interpreter image) in my K8s setup.

The title of ZEPPELIN-4154 and the first PR #3380 imply an image for each interpreter. This PR tries to solve the task.

In my opinion we should at least provide different images for the Zeppelin server and the Zeppelin interpreter. A distribution image is useful to build Zeppelin only once and copy the same version to the Zeppelin server and the Zeppelin interpreter. 

My main goal for different images is to reduce the size and start time of images in a container cluster.
The size of my current Zeppelin image is only 410.95MB. The download and total startup time of the new instance is short.
My Zeppelin interpreter image is quite large (1.53 GB). The download time is quite long.

If we want to create an image for each interpreter, the image size is reduced. All interpreter images should use the same base image to benefit from a potentially available layer.

> How it all will work on non-K8S? Like, using Docker just for not installing anything to machine

Docker is also able to set up a local network, in most cases this is done via a [bridge network](https://docs.docker.com/network/bridge/). The Zeppelin server needs access to create/modify the network via the Docker daemon's tcp interface or at least the information when new containers are created via the tcp interface.
In my opinion, a docker cluster via a docker swarm should not fall within the scope of this project.